kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -271,25 +271,67 @@ function showExamReportDetailPage(store, classId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
 }
 
-function activateExam() {
-  this.displayModal(false);
+function activateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
```

Comment:
Why do you need to dispatch this SET_EXAMS action? My assumption was that, unlike redux, since the vuex store is an observable, you can modify it in place, and any views bound to it will update automatically.